staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -444,44 +446,47 @@ func (g *Cloud) getFoundInstanceByNames(names []string) ([]*gceInstance, error)
 
 	found := map[string]*gceInstance{}
 	remaining := len(names)
+	var escapedInstanceNames []string
 
-	nodeInstancePrefix := g.nodeInstancePrefix
 	for _, name := range names {
 		name = canonicalizeInstanceName(name)
-		if !strings.HasPrefix(name, g.nodeInstancePrefix) {
-			klog.Warningf("Instance %q does not conform to prefix %q, removing filter", name, g.nodeInstancePrefix)
-			nodeInstancePrefix = ""
-		}
+		escapedInstanceNames = append(escapedInstanceNames, regexp.QuoteMeta(name))
 		found[name] = nil
 	}
 
-	for _, zone := range g.managedZones {
-		if remaining == 0 {
-			break
-		}
-		instances, err := g.c.Instances().List(ctx, zone, filter.Regexp("name", nodeInstancePrefix+".*"))
-		if err != nil {
-			return nil, err
-		}
-		for _, inst := range instances {
+	// Max instances requested per call limited to 500, so results fit on 1 page
```

Comment:
why not make the max call to list and go through the `nextToken` things to request the next pages instead of doing our own math

Reply:
reason being: what if the GCE team decides to change their page size to `200`? Now we have some weird failures as we're dropping pages